src/utils.ts

```diff
@@ -48,14 +50,55 @@ export function getConfig(): Record<string, any> {
 	);
 }
 
+/**
+ * This type is used to include and exclude the states and objects cache from the adaptert type definition depending on the creation options
+ */
+export interface AdapterInstance<
+	HasObjectsCache extends boolean | undefined = undefined,
+	HasStatesCache extends boolean | undefined = undefined
+> extends Omit<ioBroker.Adapter, "oObjects" | "oStates"> {
+	oObjects: HasObjectsCache extends true
+		? Exclude<ioBroker.Adapter["oObjects"], undefined>
+		: never;
```

Comment:
You probably want to use `undefined` here instead of `never` since currently `AdapterInstance<false>` will have `.oObject === never` which means it will be assignable to anything, as well `AdapterInstance<boolean>` will have `.oObject` unconditionally defined.  If you use `undefined` here instead it means the field will be set to undefined if the generic is not passed (or false) and will be possibly undefined when the input is boolean.
```suggestion
		: undefined;
```